### PR TITLE
Batch loop

### PR DESCRIPTION
**Pull request by jasonLaster**
So two things here:

1. i noticed that lib/watch.js was lying around unused
2. I updated the setInterval watch.js code to batch up the loop code...

---

you'll have to grep for it...

### CODE REVIEW

File: extension/js/lib/watchjs/src/watch.js [hunk 1]
```diff
@@ -461,18 +466,57 @@
...
         if(!__agent) return
 
-        __agent.lazyWorker.push({
-          context: this,
-          args: [],
-          callback: loop
-        })
 
-    }, 500);
+        var total = lengthsubjects.length;
```
Comment (jasonLaster) [inline extension/js/lib/watchjs/src/watch.js]: here's the change

**Comment by ianmstew** [comment]
Great idea on the batching within watch.js... This seems groundbreaking.  I'm wondering if this lazyworker idea could become a library, especially for extension developers?

One question.  Could changes somehow be missed by watch.js by breaking out of the event loop?

**Comment by jasonLaster** [comment]
@ianmstew i don't think we'll lose data... objects and watchers are always added... so we won't miss them

in terms of a separate lib, we should blog about it!

so guys, i pushed this change w/ breaking tests... got too excited. I'll fix minor tomorrow morning

**Comment by ianmstew** [comment]
@jasonLaster ahh, got it.  We totally should on the blog thing!

**Pull request closed without merging**